src/qibo/backends/__init__.py

```diff
@@ -28,6 +31,9 @@ def _construct_backend(name):
     else: # pragma: no cover
         # CI uses tensorflow as default backend
         K = NumpyBackend()
+        AVAILABLE_BACKENDS = [b for b in AVAILABLE_BACKENDS
+                              if "tensorflow" not in b]
+        AVAILABLE_BACKENDS.remove("custom")
```

Comment:
Also here, if I understand correctly, you are removing tf if the code is initialized with a non tensorflow backend, but this does limit the possibility to switch back to tensorflow in a later stage, right?

Reply:
This is a good point. I removed these lines so that if one uses the `QIBO_BACKEND` environment variable to set the backend to numpy, then this does not exclude potential future change to tensorflow if that is installed. Note however that in this case `defaulteinsum` and `matmuleinsum` will refer to the corresponding numpy backends, because numpy will be logged as the default backend. In order to switch to tensorflow one should then do `qibo.set_backend("tensorflow_defaulteinsum")` or just `qibo.set_backend("custom")` since custom is not available for numpy anyway.